common/txmgr/confirmer.go

```diff
@@ -297,7 +304,20 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pro
 		return fmt.Errorf("CheckConfirmedMissingReceipt failed: %w", err)
 	}
 
-	if err := ec.CheckForReceipts(ctx, head.BlockNumber()); err != nil {
+	_, latestFinalizedHead, err := ec.headTracker.LatestAndFinalizedBlock(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve latest finalized head: %w", err)
+	}
+
+	if types.Head[BLOCK_HASH](latestFinalizedHead) == nil || !latestFinalizedHead.IsValid() {
```

Comment:
I mistook the types in my previous comment for this check. I think you can do just the `IsValid` check
```suggestion
	if !latestFinalizedHead.IsValid() {
```